tests/resources/test_upload_jobs.py

```diff
@@ -172,6 +178,62 @@ def test_get_upload_job(cidc_api, clean_db, monkeypatch):
     assert res.json["id"] == other_job
 
 
+def test_requires_upload_token_auth(cidc_api, clean_db, monkeypatch):
+    """Check that the requires_upload_token_auth decorator works as expected"""
+    user_id = setup_trial_and_user(cidc_api, monkeypatch)
+    job_id = setup_upload_jobs(cidc_api)[0]
+    with cidc_api.app_context():
+        job = UploadJobs.find_by_id(job_id)
+
+    test_route = "/foobarfoo"
+
+    @requires_upload_token_auth(lambda *a, **kw: kw["upload_job"])
+    def endpoint(*args, **kwargs):
+        assert "upload_job" in kwargs
+        return "ok", 200
+
+    query_route = f"{test_route}/{job_id}"
+    nonexistent_job_id = "9999999"
+
+    # User must provide `token` query param
+    with cidc_api.test_request_context(query_route):
+        with pytest.raises(UnprocessableEntity):
```

Comment:
why not add msg matching for the `.raises`? 
goes for every `pytest.raises` below as well.